vnext/Microsoft.ReactNative/Views/ViewViewManager.cpp

```diff
@@ -530,7 +531,7 @@ void ViewViewManager::TryUpdateView(
   }
 
   // Clean up child of Control if needed
-  if (isControl && (!useControl || (hasOuterBorder != hadOuterBorder))) {
+  if (isControl && hasOuterBorder != hadOuterBorder) {
```

Comment:
Note, this would have become a bit of dead code, because a view never reverts from ViewControl to not ViewControl.